src/view/RoundView.java

```diff
@@ -120,7 +126,14 @@ public void propertyChange(PropertyChangeEvent evt) {
             JDialog dialog = optionPane.createDialog(this, submitAnswerState.getCorrectnessTitle());
             optionPane.addPropertyChangeListener(e -> {
                 if (JOptionPane.VALUE_PROPERTY.equals(e.getPropertyName())) {
-                    System.out.println("Execute post round use case here...");
+                    // Call finish round controller
+                    RoundState roundState = roundViewModel.getState();
+                    finishRoundController.execute(roundState.getGameId());
+
+                    // Update view text with new state info
+                    roundInfo.setText("Round: " + roundViewModel.getState().getCurrentRoundNumber() + "/" + roundViewModel.getState().getMaxRounds());
+                    livesInfo.setText("Lives left:" + roundViewModel.getState().getCurrentLives());
+                    genreInfo.setText("Genre: "  + roundViewModel.getState().getGenre());
```

Comment:
This is pretty minor, but I noticed that we are essentially repeating the same logic for updating the dynamic labels here as we are in the constructor. In an effort to keep our code DRY (i.e. Don't Repeat Yourself), we could probably extract this logic into a private method and then instead call that method both here and at the end of the constructor. For the constructor, this would mean we create the JLabels with an empty string (which will get properly updated once the method is called).